### PR TITLE
Remove ingame mod chooser.

### PR DESCRIPTION
**Pull request by pchote**
This implements the remaining engine changes to properly support stand-alone mod installs and removes the ability to install oramod packages to the support dir.  All that remains for #12554 after this and the other PRs are merged are straightforward packaging changes to install mod-specific launch shortcuts on each platform.

Making `Game.Mod` a required argument means that players can no longer double click OpenRA.Game.exe in their file manager and expect the game to launch.  They must instead use the launchers shipped in the installers, or for self-compiled versions either the [MD/XS gui](https://github.com/OpenRA/OpenRA/pull/12906), [the improved](https://github.com/OpenRA/OpenRA/pull/12992) `launch-game.sh`, [the new](https://github.com/OpenRA/OpenRA/pull/13033) `launch-game.cmd`, or the launch scripts in [the stand-alone mod template](https://github.com/OpenRA/OpenRAModTemplate).

### CODE REVIEW

File: OpenRA.Game/Game.cs [hunk 1]
```diff
@@ -362,10 +362,9 @@ public static void InitializeMod(string mod, Arguments args)
 
 			// Fall back to default if the mod doesn't exist
```
Comment (reaperrr) [inline OpenRA.Game/Game.cs]: This comment doesn't seem to be valid anymore.

File: mods/cnc/chrome/mainmenu.yaml [hunk 2]
```diff
@@ -78,12 +78,12 @@ Container@MENU_BACKGROUND:
...
 							Height: 35
 							Text: Extras
-						Button@MODS_BUTTON:
+						Button@CONTENT_BUTTON:
 							X: 600
 							Y: 0
 							Width: 140
 							Height: 35
-							Text: Mods
+							Text: Game Content
```
Comment (reaperrr) [inline mods/cnc/chrome/mainmenu.yaml]: Nit: Wouldn't it be better/more consistent to call this `Mod Content`?

File: mods/common/chrome/mainmenu.yaml [hunk 3]
```diff
@@ -70,12 +70,12 @@ Container@MAINMENU:
...
 							Text: Extras
 							Font: Bold
-						Button@MODS_BUTTON:
+						Button@CONTENT_BUTTON:
 							X: PARENT_RIGHT/2-WIDTH/2
 							Y: 220
 							Width: 140
 							Height: 30
-							Text: Mods
+							Text: Game Content
```
Comment (reaperrr) [inline mods/common/chrome/mainmenu.yaml]: Same here.

File: mods/d2k/chrome/mainmenu.yaml [hunk 4]
```diff
@@ -57,12 +57,12 @@ Container@MAINMENU:
...
 							Text: Extras
 							Font: Bold
-						Button@MODS_BUTTON:
+						Button@CONTENT_BUTTON:
 							X: PARENT_RIGHT/2-WIDTH/2
 							Y: 220
 							Width: 140
 							Height: 30
-							Text: Mods
+							Text: Game Content
```
Comment (reaperrr) [inline mods/d2k/chrome/mainmenu.yaml]: Same here.

**Comment by pchote** [comment]
Changed button labels to "Manage Content" and rebased.

**Comment by pchote** [comment]
I've added a commit that fixes the deb packaging.  It looks like our packaging scripts got out out sync across the last set of parallel PRs.

**Comment by pchote** [comment]
Rebased after #12828.

**Review by reaperrr — APPROVED** [review]

**Comment by atlimit8** [comment]
Has this been tested by a reviewer in Windows?

**Comment by reaperrr** [comment]
> Has this been tested by a reviewer in Windows?

Yes (me).

**Comment by pchote** [comment]
> > Has this been tested by a reviewer in Windows?
> 
> Yes (me).

I have also tested this on all three platforms as part of #13129, #13133, #13138.

Also, rebased after #13141.

File: OpenRA.Mods.Common/Widgets/Logic/ModBrowserLogic.cs [hunk 5]
```diff
@@ -115,9 +115,7 @@ public ModBrowserLogic(Widget widget, ModData modData)
 				catch (Exception) { }
 			}
 
-			Manifest initialMod;
-			Game.Mods.TryGetValue(Game.Settings.Game.PreviousMod, out initialMod);
-			SelectMod(initialMod != null && initialMod.Id != "modchooser" ? initialMod : Game.Mods["ra"]);
+			SelectMod(Game.Mods["ra"]);
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Widgets/Logic/ModBrowserLogic.cs]: How can we be sure that this is installed?

File: OpenRA.Server/Program.cs [hunk 6]
```diff
@@ -30,14 +30,17 @@ static void Main(string[] args)
...
+			if (modID != null && (File.Exists(modID) || Directory.Exists(modID)))
 			{
-				explicitModPaths = new[] { modArgument };
-				arguments.ReplaceValue("Game.Mod", Path.GetFileNameWithoutExtension(modArgument));
+				explicitModPaths = new[] { modID };
+				modID = Path.GetFileNameWithoutExtension(modID);
 			}
 
+			if (modID == null)
+				throw new ArgumentException("The mod to launch was not specified", "mod");
```
Comment (obrakmann) [inline OpenRA.Server/Program.cs]: How about "Game.Mod argument missing or mod could not be found"? It's a bit clearer, and gives the user the info how to fix the error.

File: OpenRA.Game/Game.cs [hunk 7]
```diff
@@ -362,10 +362,9 @@ public static void InitializeMod(string mod, Arguments args)
...
 			if (mod == null)
-				mod = args.GetValue("Engine.DefaultMod", "modchooser");
+				throw new ArgumentException("The mod to launch was not specified", "mod");
```
Comment (obrakmann) [inline OpenRA.Game/Game.cs]: How about "Game.Mod argument missing or mod could not be found"? It's a bit clearer, and gives the user the info how to fix the error.

**Review by obrakmann — COMMENTED** [review]
Looks fine so far, just a couple of minor things. Package builds as well, and its contents look fine.

File: OpenRA.Mods.Common/Widgets/Logic/ModBrowserLogic.cs (hunk 5 above)
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/ModBrowserLogic.cs]: This was a temporary stopgap.  The entire file gets removed in the next commit :smile:

**Comment by pchote** [comment]
Ok. Changed.

**Review by obrakmann — APPROVED** [review]
👍

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/9518ba66f9ee83848e7617eeb7d387950e9af41c)

**Pull request merged**